File: Flutter/HUMAN_Demo/ios/HumanDemo/HumanManager.swift
import Foundation
import Flutter
import HUMAN

class HumanManager {
    
    static let shared = HumanManager()
    
    func start() {
        do {
            let policy = HSPolicy()
            policy.automaticInterceptorPolicy.interceptorType = .none
            policy.doctorAppPolicy.enabled = true
            try HumanSecurity.start(appId: "PXj9y4Q8Em", policy: policy)
        }
        catch {
            print("error: \(error)")
        }
    }
    
    func setupChannel(with controller: FlutterViewController) {
        let humanChannel = FlutterMethodChannel(name: "com.humansecurity.demo/human", binaryMessenger: controller.binaryMessenger)
        humanChannel.setMethodCallHandler({(call: FlutterMethodCall, result: @escaping FlutterResult) -> Void in
            if call.method == "_getHumanHeaders" {
                var json: String?
                do {
                    let headers = HumanSecurity.BD.headersForURLRequest(forAppId: nil)
                    let data = try JSONSerialization.data(withJSONObject: headers)
                    json = String(data: data, encoding: .utf8)
                }
                catch {
                    print("error: \(error)")
                }
                result(json)
            }
            else if call.method == "_handleHumanResponse" {
                if let response = call.arguments as? String, let data = response.data(using: .utf8), let httpURLResponse = HTTPURLResponse(url: URL(string: "https://fake.url.com")!, statusCode: 403, httpVersion: nil, headerFields: nil) {
                    let handled = HumanSecurity.BD.handleResponse(response: httpURLResponse, data: data) { challengeResult in
                        result(challengeResult == .solved ? "solved" : "cancelled")
                    }
                    if handled {
                        return
                    }
                }
                result("false")
            }
            else {
                result("")
            }
        })
    }
}

File: Flutter/HUMAN_Demo/android/app/src/main/kotlin/com/humansecurity/demo/HumanManager.kt
package com.perimeterx.perimeterx

import android.app.Application
import com.humansecurity.mobile_sdk.HumanSecurity
import com.humansecurity.mobile_sdk.main.HSBotDefenderChallengeResult
import com.humansecurity.mobile_sdk.main.policy.HSAutomaticInterceptorType
import com.humansecurity.mobile_sdk.main.policy.HSPolicy
import com.humansecurity.mobile_sdk.main.policy.HSStorageMethod
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import org.json.JSONObject

class HumanManager {

    companion object {

        fun start(application: Application) {
            try {
                val policy = HSPolicy()
                policy.storageMethod = HSStorageMethod.DATA_STORE
                policy.automaticInterceptorPolicy.interceptorType = HSAutomaticInterceptorType.NONE
                policy.doctorAppPolicy.enabled = true
                HumanSecurity.start(application, "PXj9y4Q8Em", policy)
            }
            catch (exception: Exception) {
                println("failed to start. exception: $exception")
            }
        }

        fun handleEvent(call: MethodCall, result: MethodChannel.Result) {
            if (call.method == "_getHumanHeaders") {
                val json = JSONObject(HumanSecurity.BD.headersForURLRequest(null) as Map<*, *>?)
                result.success(json.toString())
            }
            else if (call.method == "_handleHumanResponse") {
                val handled = HumanSecurity.BD.handleResponse(call.arguments!! as String) { challengeResult: HSBotDefenderChallengeResult ->
                    result.success(if (challengeResult == HSBotDefenderChallengeResult.SOLVED) "solved" else "cancelled")
                    null
                }
                if (!handled) {
                    result.success("false")
                }
            }
            else {
                result.notImplemented()
            }
        }
    }
}
